example/main.tf

```diff
@@ -52,60 +53,58 @@ resource "google_compute_subnetwork" "vpc_subnetwork" {
   # a dash, lowercase letter, or digit, except the last character, which
   # cannot be a dash.
   #name = "default-${var.gcp_cluster_region}"
-  name = "${var.vpc_subnetwork_name}"
+  name = var.vpc_subnetwork_name
 
-  ip_cidr_range = "${var.vpc_subnetwork_cidr_range}"
+  ip_cidr_range = var.vpc_subnetwork_cidr_range
 
   # The network this subnet belongs to. Only networks that are in the
   # distributed mode can have subnetworks.
-  network = "${var.vpc_network_name}"
+  network = var.vpc_network_name
 
   # An array of configurations for secondary IP ranges for VM instances
   # contained in this subnetwork. The primary IP of such VM must belong to the
   # primary ipCidrRange of the subnetwork. The alias IPs may belong to either
   # primary or secondary ranges.
-  secondary_ip_range = [
-    {
-      range_name    = "${var.cluster_secondary_range_name}"
-      ip_cidr_range = "${var.cluster_secondary_range_cidr}"
-    },
-    {
-      range_name    = "${var.services_secondary_range_name}"
-      ip_cidr_range = "${var.services_secondary_range_cidr}"
-    },
-  ]
+  secondary_ip_range {
+    range_name    = var.cluster_secondary_range_name
+    ip_cidr_range = var.cluster_secondary_range_cidr
+  }
+  secondary_ip_range {
+    range_name    = var.services_secondary_range_name
+    ip_cidr_range = var.services_secondary_range_cidr
+  }
```

Comment:
I found this code a little hard to understand at first. I guess it's declarative so it's executing a method twice, but I read it as a config that is overwriting itself.  I would love to get a 2nd opinion.  The updated code does function correctly. 👍

Reply:
Yeah it's slightly confusing to read but makes sense. Possibly update the comment above where it refers to an array to say that it creates two separate `secondary_ip_range`s.